.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,107 @@
+on:
+  workflow_call:
+    inputs:
+      secret_registry:
+        type: boolean
+        description: All registry related inputs should be treated as secret values
+        required: true
+
+      image:
+        type: string
+        description: Static image value for the build
+        required: true
+
+      password:
+        type: string
+        description: Registry password key to lookup in secrets
+        required: true
+
+      username:
+        type: string
+        description: Username for the registry login
+        required: true
+
+      registry:
+        type: string
+        description: Destination registry for image push
+        required: true
+
+
+      tag:
+        type: string
+        description: Tag for the built image
+        required: true
+
+      arch:
+        type: string
+        description: Architecture for the image
+        required: true
+
+      org:
+        type: string
+        description: Organization part of the image name
+        required: false
+        default: "rancher-sandbox"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      packages: write
+    outputs:
+      digest: ${{ steps.image.outputs.digest }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build the image
+        id: image
+        uses: ./.github/workflows/release_build
+        with:
+          arch: ${{ inputs.arch }}
+          tag: ${{ inputs.tag }}
+          org: ${{ inputs.org }}
+          registry: ${{ inputs.secret_registry && secrets[inputs.registry] || inputs.registry }}
+          username: ${{ inputs.secret_registry && secrets[inputs.username] || inputs.username }}
+          password: ${{ secrets[inputs.password] }}
+
+  sign:
+    runs-on: ubuntu-latest
+    needs: [build]
+    permissions:
+      actions: read
+      id-token: write
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Sign image with cosign
+        uses: ./.github/workflows/release_sign
+        with:
+          image: ${{ inputs.secret_registry && secrets[inputs.image] || inputs.image }}-${{ inputs.arch }}
+          digest: ${{ needs.build.outputs.digest }}
+          identity: https://github.com/${{ inputs.org }}/rancher-turtles/.github/workflows/release.yaml@refs/tags/${{ inputs.tag }}
+          oids-issuer: https://token.actions.githubusercontent.com
+          registry: ${{ inputs.secret_registry && secrets[inputs.registry] || inputs.registry }}
+          username: ${{ inputs.secret_registry && secrets[inputs.username] || inputs.username }}
+          password: ${{ secrets[inputs.password] }}
+
+  # Changes from https://github.com/slsa-framework/slsa-github-generator/pull/2918 are required to make the provenance work for production registry
```

Comment:
Referred changes seem to be merged, is the plan to enable the provenance not in this PR but later?

Reply:
Yes, it was merged but was not released. I’ve switched the image to be a collected from vars as was [recommended](https://github.com/slsa-framework/slsa-github-generator/issues/2917#issuecomment-1783508999), as it is not a concern to leave image name in logs for us, and recor transparency log will still contain the full image name.

Reply:
We'll leave the provenance commented out until we fully decide what we are going to do.

Reply:
Which remind me i need to create an issue

Reply:
> We'll leave the provenance commented out until we fully decide what we are going to do.
Looks like we enabled it in the same PR..

Reply:
Yes, it seems Danil uncommented provenance after this comment.

Reply:
Yes, the data is stored in the variable, and is passed to the job, avoiding scenarios when the job output is redacted.

Reply:
@Danil-Grigorev @furkatgofurov7 - can we resolve this conversation? So this PR can be merged.